src/main/java/scmspain/karyon/restrouter/RestRouterHandler.java

```diff
@@ -0,0 +1,222 @@
+package scmspain.karyon.restrouter;
+
+import com.google.common.net.HttpHeaders;
+import com.google.common.net.MediaType;
+import io.netty.buffer.ByteBuf;
+import io.reactivex.netty.protocol.http.server.HttpServerRequest;
+import io.reactivex.netty.protocol.http.server.HttpServerResponse;
+import io.reactivex.netty.protocol.http.server.RequestHandler;
+import org.apache.commons.lang.StringUtils;
+import org.commonjava.mimeparse.MIMEParse;
+import rx.Observable;
+import scmspain.karyon.restrouter.exception.CannotSerializeException;
+import scmspain.karyon.restrouter.exception.InvalidAcceptHeaderException;
+import scmspain.karyon.restrouter.handlers.DefaultErrorHandler;
+import scmspain.karyon.restrouter.handlers.ErrorHandler;
+import scmspain.karyon.restrouter.handlers.RestRouterErrorDTO;
+import scmspain.karyon.restrouter.serializer.SerializeManager;
+import scmspain.karyon.restrouter.serializer.SerializeWriter;
+import scmspain.karyon.restrouter.serializer.Serializer;
+import scmspain.karyon.restrouter.transport.http.RestUriRouter;
+import scmspain.karyon.restrouter.transport.http.Route;
+import scmspain.karyon.restrouter.transport.http.RouteNotFound;
+
+import javax.inject.Inject;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class RestRouterHandler implements RequestHandler<ByteBuf, ByteBuf> {
```

Comment:
Maybe I'm wrong, but you probably should do a git move before changing it to be able to see the differences, now I don't know if you have changed everything or nothing.
Mostly in that kind of PR whit 40 files changed, which is really a lot to be reviewed, you should try to simplify the job.